src/UI/Components/InstanceProvider/InstanceProvider.tsx

```diff
@@ -1,34 +1,79 @@
 import React, { useContext } from "react";
 import { ServiceModel } from "@/Core";
-import { DependencyContext } from "@/UI";
-import { RemoteDataView } from "@/UI/Components";
+import { useGetInstanceWithRelations } from "@/Data/Managers/V2/GetInstanceWithRelations";
+import { useGetMetadata } from "@/Data/Managers/V2/GetMetadata";
+import { DependencyContext, words } from "@/UI";
+import { ErrorView, LoadingView } from "@/UI/Components";
 import Canvas from "@/UI/Components/Diagram/Canvas";
 
+/**
+ * Renders the InstanceProvider component.
+ * It serves purpose to not provide all the neccesary data for the Canvas at the same time to avoid unnecessary rerenders,
```

Comment:
```suggestion
 * It serves purpose to not provide all the necessary data for the Canvas at the same time to avoid unnecessary rerenders,
```

The text confuses me a bit, why would it prevent serving necessary data?